hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/api/OMDBInsightEndpoint.java

```diff
@@ -627,6 +627,28 @@ private void getPendingForDeletionDirInfo(
     }
   }
 
+  private void calculateTotalPendingDeletedDirSizes(Map<String, Long> dirSummary) {
+    long totalUnreplicatedSize = 0L;
+    long totalReplicatedSize = 0L;
+
+    Table<String, OmKeyInfo> deletedDirTable = omMetadataManager.getDeletedDirTable();
+    try (TableIterator<String, ? extends Table.KeyValue<String, OmKeyInfo>> iterator = deletedDirTable.iterator()) {
+      while (iterator.hasNext()) {
+        Table.KeyValue<String, OmKeyInfo> kv = iterator.next();
+        OmKeyInfo omKeyInfo = kv.getValue();
+        if (omKeyInfo != null) {
+          totalUnreplicatedSize += fetchSizeForDeletedDirectory(omKeyInfo.getObjectID());
```

Comment:
based on previous PR, need handle this

Reply:
I have updated the _TODO_ for handling the replicated size calculation based on the PR: https://github.com/apache/ozone/pull/8568 post merge.